python/pysmurf/client/command/smurf_command.py

```diff
@@ -408,12 +487,132 @@ def get_number_processed_channels(self, band=None, **kwargs):
         n_processed_channels=int(0.8125*n_channels)
         return n_processed_channels
 
-    def set_defaults_pv(self, **kwargs):
-        """
-        Sets the default epics variables
-        """
-        self._caput(self.epics_root + ':AMCc:setDefaults', 1, wait_after=30,
-            **kwargs)
+    def set_defaults_pv(self, wait_after_sec=30.0,
+                        max_timeout_sec=400.0, caget_timeout_sec=10.0,
+                        **kwargs):
+        r"""Loads the default configuration.
+
+        Calls the rogue `setDefaults` command, which loads the default
+        software and hardware configuration.
+
+        If using pysmurf core code versions >=4.1.0 (as reported by
+        :func:`get_pysmurf_version`), returns `True` if the
+        `setDefaults` command was successfully executed on the rogue
+        side, or failed.  Returns `None` for older versions.
+
+        Args
+        ----
+        wait_after_sec : float or None, optional, default 30.0
+            If not None, the number of seconds to wait after
+            triggering the rogue `setDefaults` command.
+        max_timeout_sec : float, optional, default 400.0
+            Seconds to wait for system to configure before giving up.
+            Only used for pysmurf core code versions >= 4.1.0.
+        caget_timeout_sec : float, optional, default 10.0
+            Seconds to wait for each poll of the configuration process
+            status registers (see :func:`get_configuring_in_progress`
+            and :func:`get_system_configured`).  Only used for pysmurf
+            core code versions >= 4.1.0.
+        \**kwargs
+            Arbitrary keyword arguments.  Passed directly to
+            all `_caget` calls.
+
+        Returns
+        -------
+        success : bool or None
+            Returns `True` if the system was successfully configured,
+            otherwise returns `False`.  Configuration checking is only
+            implemented for pysmurf core code versions >= 4.1.0.  If
+            configuration validation is not available, returns None.
+
+        See Also
+        --------
+        :func:`get_configuring_in_progress` : Whether or not
+                configuration process in progress.
+        :func:`get_system_configured` : Returns final state of
+                configuration process.
+
+        """
+        # strip any commit info off the end of the pysmurf version
+        # string
+        pysmurf_version = self.get_pysmurf_version(**kwargs).split('+')[0]
```

Comment:
@swh76 why do you read the pysmurf version register each time this command is call? Shouldn't this be done the the `pysmurf.client.SmurfControl` is instantiated, and assumes it remains the same? 

Unless you are planning to support restarting the rogue server and keep the same pysmurf client session running?

Reply:
That is why but I agree caching is smarter.  I'll add something for that ; wait to push.

Reply:
@jesusvasquez333 ok sounds good, I'll leave as is, at least for now.  I started a new issue to think about this if we want to revisit in the future - https://github.com/slaclab/pysmurf/issues/476.